src/translator/quality_estimator.h

```diff
@@ -0,0 +1,46 @@
+#ifndef SRC_BERGAMOT_QUALITY_ESTIMATOR_H_
+#define SRC_BERGAMOT_QUALITY_ESTIMATOR_H_
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "annotation.h"
+#include "intgemm/intgemm.h"
+#include "response.h"
+
+namespace marian {
+namespace bergamot {
+
+class QualityEstimator {
+ private:
+  struct WordFeatures {
+    float bpe_sum_over_len;
+    float min_bpe;
+    float len_bpe;
+    float overall_mean;
+  };
```

Comment:
> Since you already have intgemm, it might be worth considering taking in B x D matrix X where B = batch-size, D = featureDim for features corresponding to all sentences in a Request/Response in a single batch, use the mm to do more efficient multiply (unsure how good this will be though if preprocessing is sequential and the bottleneck...).

This would be a good thing to do. However, since the model takes words as features, such operation ` B x  D` would be actually `B x S x D`, where `S` is the word level length of the longest sentence in batch and hence, multiplying such tensor seems a little bit tricky. Therefore, for this deliverable, I have decided to work using only a single sentence and just analyze such complex scenario in the future